### PR TITLE
add python3 compatibility

### PR DESCRIPTION
**Pull request by Rechi**

### CODE REVIEW

**Comment by Rechi** [comment]
@tamland ping

File: lib/routing.py [hunk 1]
```diff
@@ -104,15 +114,22 @@ def add_route(self, func, pattern):
...
     def redirect(self, path):
         self._dispatch(path)
 
     def _dispatch(self, path):
-        for view_func, rules in self._rules.iteritems():
+        try:
+            items = self._rules.iteritems()
+        except AttributeError:
+            items = iter(self._rules.items())
+        for view_func, rules in items:
```
Comment (tamland) [inline lib/routing.py]: Just use items() for both. this list is not big

File: lib/routing.py [hunk 2]
```diff
@@ -104,15 +114,22 @@ def add_route(self, func, pattern):
 
     def run(self, argv=sys.argv):
         if len(argv) > 2:
-            self.args = parse_qs(argv[2].lstrip(b'?'))
+            try:
+                self.args = parse_qs(argv[2].lstrip(b'?'))
+            except TypeError:
+                self.args = parse_qs(argv[2].lstrip('?'))
```
Comment (tamland) [inline lib/routing.py]: Not sure why the `b` is there, it doesn't seem to be necessary. So this `try` can be removed as well

**Comment by tamland** [comment]
Sorry didn't see this. After the github update a couple months ago I don't get any repository notifications anymore (even though I am 'watching') :\

File: lib/routing.py [hunk 3]
```diff
@@ -63,7 +69,11 @@ def route_for(self, path):
         if path.startswith(self.base_url):
             path = path.split(self.base_url, 1)[1]
 
-        for view_fun, rules in self._rules.iteritems():
```
Comment (Rechi) [inline lib/routing.py:66]: Does this iterate over the same items as the one in `_dispatch`? If yes, should I also just use `items()`?

Comment (tamland) [reply lib/routing.py:66]: yes!

**Comment by Rechi** [comment]
@tamland updated

**Comment by tamland** [comment]
thanks!

**Comment by Rechi** [comment]
Can you push an update to https://github.com/xbmc/repo-scripts/tree/gotham/script.module.routing?

**Comment by Rechi** [comment]
@tamland you missed my request to updated the script in the repo?

**Comment by tamland** [comment]
Yeah, sorry I was low on time and forgot. I'll try to remember it. I've added you as collaborator if you're in a hurry

**Comment by Rechi** [comment]
Thanks for adding me as collaborator, but I don't understand how that helps updating repo-scripts?

Just noticed this again as I did some python3 building/testing for windows.
I'm not in a hurray, but devs using your script can't make their addon py3 compatible.

**Comment by tamland** [comment]
I just meant feel free to do it. You have access to bumping version and pushing to add-on repo right? Anyway, it's updated now.

**Pull request merged**